earth_enterprise/src/fusion/portableglobe/cutter/cgi-bin/globe_cutter_app.py

```diff
@@ -344,6 +346,38 @@ def AppendInfoFile(self, message):
     fp.write("%s\n\n" % message)
     fp.close()
 
+  def HttpGet(self, url):
+    """Make an HTTP or HTTPS request with a context for checking or ignoring
+    certificate errors depending on Config settings.
+    """
+    context = None
+    response_data = ""
+    http_status_code = 0
+
+    try:
+      # Set the context based on cert requirements
+
+      if CONFIGS.GetBool("VALIDATE_CERTIFICATE"):
+        cert_file = CONFIGS.GetStr("CERTIFICATE_CHAIN_PATH")
+        key_file = CONFIGS.GetStr("CERTIFICATE_KEY_PATH")
+        context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
+        context.load_cert_chain(cert_file, keyfile=key_file)
+      else:
+        context = ssl.create_default_context()
+        context.check_hostname = False
+        context.verify_mode = ssl.CERT_NONE
+
+      fp = urllib2.urlopen(url, context=context)
+      http_status_code = fp.getcode()
+      response_data = fp.read()
+      fp.close()
```

Comment:
Is it possible to use the `with` statement to auto close this?  If the read raised an exception would we leak the open?